### DISCUSSION TITLE
Simulation time

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#28729] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,
Is there a way to output the simulation time after each iteration? I know that PerfGraphOutput provides the total time for the entire simulation, but I would like to get timing information during the simulation. For example, if I have 100 steps, I want to know how long it takes to complete the 10th, 20th, ..., up to the 100th time step, not just the overall simulation time.

**Comment by GiudGiud** [comment]
Hello

You can output specific items from the perf graph at every time step using this postprocessor
https://mooseframework.inl.gov/source/postprocessors/PerfGraphData.html

Guillaume

**Reply by travismui** [reply]
Was just going to suggest this too, you could use something like this to output the current simulation time at each timestep and use that to determine how long each step took to simulate:

```
[Postprocessors]
  [run_time]
    type = PerfGraphData
    section_name = Root
    data_type = TOTAL
  []
[]
 ```

**Reply by alimostafavi24** [reply]
Thanks for the suggestions @GiudGiud @travismui . This one works perfectly.
